### PR TITLE
[Couenne] add initial builder

### PR DESCRIPTION
**Pull request by odow**
Follow-up to https://github.com/JuliaPackaging/Yggdrasil/pull/2730, https://github.com/JuliaPackaging/Yggdrasil/pull/2770, and https://github.com/JuliaPackaging/Yggdrasil/pull/2774.

### CODE REVIEW

File: C/Coin-OR/Couenne/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    --with-bonmin-lib="-lCoinUtils -lClp -lCgl -lCbc -lbonmin -lbonminampl"
+
+make
+make install
+"""
+
+platforms = supported_platforms()
+platforms = filter!(!Sys.isfreebsd, platforms)
+platforms = expand_cxxstring_abis(platforms)
+platforms = expand_gfortran_versions(platforms)
```
Comment (giordano) [inline C/Coin-OR/Couenne/build_tarballs.jl:47]: Why?  As far as I can see from the log the resulting lib doesn't link to libgfortran.

Comment (odow) [reply C/Coin-OR/Couenne/build_tarballs.jl:47]: I just started with the same bonmin builder. If it depends on a JLL that expands gfortran, do we not also need to expand?

Comment (giordano) [reply C/Coin-OR/Couenne/build_tarballs.jl:47]: Nope.  Only if the library itself actually links to libgfortran.  Generally speaking, what I do is to not expand anything initially and just look for warnings during audit.

Comment (giordano) [reply C/Coin-OR/Couenne/build_tarballs.jl:47]: Which is also suggested at the end of this section of the manual: https://docs.binarybuilder.org/stable/building/#Platforms

**Comment by odow** [comment]
Okay, I think this is good by me.

File: C/Coin-OR/Couenne/bundled/patches/fixes.patch [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
++inline bool is_expr_zero (expr* e) {
++  if (e==NULL) {
++    return true;
++  }
++  size_t op = (size_t)(e->op);
++  return (op == OPNUM) && (fabs(((expr_n *)e) -> v)  < COUENNE_EPS);
++}
+ 
+ void createCommonExpr (CouenneProblem *p, const ASL *asl, int i, int which);
+
```
Comment (matbesancon) [inline C/Coin-OR/Couenne/bundled/patches/fixes.patch:79]:
```suggestion

```

Comment (matbesancon) [reply C/Coin-OR/Couenne/bundled/patches/fixes.patch:79]: there was a space on the last line, was wondering why we didn't see a new line :sweat_smile:

**Pull request merged**